### ISSUE TITLE
Open a connection suddenly closed by server

### ISSUE THREAD

**Issue by camillo777** [theturtle32/WebSocket-Node#202]
Hi,
I am using the "Server example" on an OpenShift node.js implementation, but when I try to connect (I tried several sources: Android, JSFiddle, www.websocket.org) and they always says "Connect" and "Disconnect" as well.

Nothing is logged inside the Node.js logs apart from:
Fri May 29 2015 07:06:54 GMT-0400 (EDT) Server is listening on 127.13.35.129 : 8000

This is the exact code I am using:

var ipaddress = process.env.OPENSHIFT_NODEJS_IP || "127.0.0.1";
var port      = 8000;

var WebSocketServer = require('websocket').server;

var http = require('http');

var server = http.createServer(function(request, response) {
                               console.log((new Date()) + ' Received request for ' + request.url);
                               //response.writeHead(404);
                               //response.end();
                               });
server.listen(port, ipaddress, function() {
              console.log((new Date()) + ' Server is listening on '+ipaddress+' : '+port);
              });

wsServer = new WebSocketServer({
                               httpServer: server,
                               // You should not use autoAcceptConnections for production
                               // applications, as it defeats all standard cross-origin protection
                               // facilities built into the protocol and the browser.  You should
                               // *always* verify the connection's origin and decide whether or not
                               // to accept it.
                               autoAcceptConnections: true
                               });

function originIsAllowed(origin) {
    // put logic here to detect whether the specified origin is allowed.
    return true;
}

wsServer.on('request', function(request) {
            console.log((new Date()) + ' request recv');
            
            if (!originIsAllowed(request.origin)) {
            // Make sure we only accept requests from an allowed origin
            request.reject();
            console.log((new Date()) + ' Connection from origin ' + request.origin + ' rejected.');
            return;
            }
            
//            var connection = request.accept('echo-protocol', request.origin);
            var connection = request.accept(null, request.origin);
  
            console.log((new Date()) + ' Connection accepted.');
            connection.on('message', function(message) {
                          if (message.type === 'utf8') {
                          console.log('Received Message: ' + message.utf8Data);
                          connection.sendUTF(message.utf8Data);
                          }
                          else if (message.type === 'binary') {
                          console.log('Received Binary Message of ' + message.binaryData.length + ' bytes');
                          connection.sendBytes(message.binaryData);
                          }
                          });
            connection.on('close', function(reasonCode, description) {
                          console.log((new Date()) + ' Peer ' + connection.remoteAddress + ' disconnected.');
                          });
            });

**Comment by theturtle32** [comment]
I'm guessing there's something up with OpenShift's WebSocket reverse proxy layer. I've had a few other issues opened by users specifically with OpenShift integration questions about things that work fine for everybody else...

**Comment by theturtle32** [comment]
Have you tried this on something other than openshift? can you provide more information? If so, reopen this issue.

**Comment by camillo777** [comment]
Hi theturtle32,
yes I have tried on "modulus.io" and it works like a charme; for completeness I have also tried "socket.io" and it has exactly the same problem of Your library, so it is not definitely a problem with your library.
The problem is I do not know how to contact OpenShift people, I am not so powerful to be heard by RedHat people, I have already added a post on StackExchange without any help. I am on a free trial with OpenShift, but I do not want to pay for something that does not work.
I do not know if You have more power than me! :-)
